pulsar-client/pom.xml

```diff
@@ -116,6 +116,11 @@
       <groupId>com.fasterxml.jackson.module</groupId>
       <artifactId>jackson-module-jsonSchema</artifactId>
     </dependency>
+
+    <dependency>
+      <groupId>joda-time</groupId>
+      <artifactId>joda-time</artifactId>
```

Comment:
Should we also include in the shaded jar

Reply:
joda-time lib is used by logical types in pulsar avro schema. Considering logical types is user-facing types, so not included in the shaded jar.

Reply:
Makes sense. Would it be possible to have it as an optional dependency then? In the same way as is Protobuf for pulsar-client? 

For Protobuf, the reason is that if application uses Protobuf they will already have that in their classpath. Same for Joda, if application wants to get the schema of Joda date, they'll already have that.

If we make it optional, it would have less chances to conflict with other versions in the application classpath.

Reply:
I'm already make it optional.